### ISSUE TITLE
Object scale & noobject scale.

### ISSUE THREAD

**Issue by p058** [AlexeyAB/darknet#559]
Why were object/noobject scale dropped in Yolov3? Especially for the last(small scale) conv layer of grid size 52 x 52? won't the loss from noobject grid cells overpower the loss from those of grid cells?

 (from yolov1 paper : "This pushes the “confidence” scores of those cells towards zero, often overpowering the gradient from cells that do contain objects. This can lead to model instability, causing training to diverge early on.")

When I am trying to train yolov3 in pytorch without object scales, I see that the confidence scores values go to zero for the small scale predictions, I figured it could be because of a large number of grid cells (52 x 52) have no objects. I am just trying to understand the thought process. Will appreciate any insights. 

Thanks and great work @AlexeyAB with maintaining the repo and detailed answers to all the issuses & questions.

**Comment by AlexeyAB** [comment]
Yes, in Yolo v3 these params are absent: https://github.com/AlexeyAB/darknet/blob/1b85111217e90bc20907070283d5de9d9d750057/cfg/yolov2.cfg#L251-L252

I do not know if it was really this problem, or there was another problem in the Yolo in 1, such as - each cell had only one t0 (confidence) for 2 anchors (boxes), so anchor without object can overpower the gradient from anchor with object.


* What object size do you use relative to image size?

* You can try to add `object_scale=5` to this code for case when **there is object**, so will be 
`l.delta[obj_index] = 5*(1 - l.output[obj_index]);` and try to measure mAP increasing.

----

Yolo v3:

* if there isn't object: https://github.com/AlexeyAB/darknet/blob/1b85111217e90bc20907070283d5de9d9d750057/src/yolo_layer.c#L189
* if **there is object**: https://github.com/AlexeyAB/darknet/blob/1b85111217e90bc20907070283d5de9d9d750057/src/yolo_layer.c#L234

----

In the Yolo v2 were:
* if there is no object: https://github.com/AlexeyAB/darknet/blob/1b85111217e90bc20907070283d5de9d9d750057/src/region_layer.c#L266
* if there is object: https://github.com/AlexeyAB/darknet/blob/1b85111217e90bc20907070283d5de9d9d750057/src/region_layer.c#L338
